CorsixTH/Lua/game_ui.lua

```diff
@@ -750,6 +752,29 @@ function GameUI:onMouseWheel(x, y)
   return UI.onMouseWheel(self, x, y)
 end
 
+--! Announcements to the player.
+--!param msgs (array of string) Messages to select from.
+--!param rnd_frac (optional float in range (0, 1]) Fraction of times that the
+--    call actually says something.
+--!param priority (optional string - Low, Normal, High, Critical) Priority of announcement
+--!return (boolean) Whether a message was given to the user.
+function GameUI:makeAnnouncement(msgs, rnd_frac, priority)
```

Comment:
This name seems wrong; you're trying to distinguish from playAnnouncement below, but the difference isn't that the announcement is made, it's that there is an element of chance involved.

rnd_frac should be after priority, matching the name, priority order of the function it delgates to. Either msgs or names is named wrong because msgs is an array of names.

Perhaps `playRandomAnnouncement(names, priority, played_callback, played_callback_delay, chance_to_play)`